src/main/java/mcp/mobius/waila/overlay/tooltiprenderers/TTRenderRFBar.java

```diff
@@ -0,0 +1,133 @@
+package mcp.mobius.waila.overlay.tooltiprenderers;
+
+import java.awt.Dimension;
+import java.text.NumberFormat;
+import java.util.function.Function;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.util.ResourceLocation;
+
+import org.lwjgl.opengl.GL11;
+
+import mcp.mobius.waila.api.IWailaCommonAccessor;
+import mcp.mobius.waila.api.IWailaVariableWidthTooltipRenderer;
+import mcp.mobius.waila.overlay.DisplayUtil;
+
+public class TTRenderRFBar implements IWailaVariableWidthTooltipRenderer {
+
+    int maxStringW;
+
+    private final Function<Integer, String> formatNumber;
+    private static final NumberFormat energyFormat = NumberFormat.getInstance();
+    private static final int height = 12;
+    private static final int width = 2;
+
+    static {
+        energyFormat.setGroupingUsed(true);
+    }
+
+    public TTRenderRFBar() {
+        formatNumber = energyFormat::format;
+    }
+
+    @Override
+    public Dimension getSize(String[] params, IWailaCommonAccessor accessor) {
+        int commaCount = (int) (Math.floor((params[0].length() - 1) / 3D) + Math.floor((params[1].length() - 1) / 3D));
```

Comment:
i would still prefer a simpler way to get size, and i even have a new reason for that: separator symbol can be different and have different width too